include/uapi/sound/sof/header.h

```diff
@@ -46,14 +46,14 @@ struct sof_manifest_tlv {
  * @abi_minor: Minor ABI version
  * @abi_patch: ABI patch
  * @count: count of tlv items
- * @items: consecutive variable size tlv items
+ * @items: consecutive variable size tlv items (struct sof_manifest_tlv)
  */
 struct sof_manifest {
 	__le16 abi_major;
 	__le16 abi_minor;
 	__le16 abi_patch;
 	__le16 count;
-	struct sof_manifest_tlv items[];
+	__u8 items[];
```

Comment:
sparse can be silenced with the following alternative:
```diff
diff --git a/include/uapi/sound/sof/header.h b/include/uapi/sound/sof/header.h
index f125f7772ee7..a60cbee18ecb 100644
--- a/include/uapi/sound/sof/header.h
+++ b/include/uapi/sound/sof/header.h
@@ -37,7 +37,7 @@ struct sof_abi_hdr {
 struct sof_manifest_tlv {
 	__le32 type;
 	__le32 size;
-	__u8 data[];
+	__u8 data[0];
 };
```

Reply:
this is deprecated in kernel code, you will not be able to upstream this @ujfalusi

Reply:
I just don't know why we insist with the [] notation, what's wrong with making this a pointer and move on?

Reply:
It is not a pointer, it is a possible continuation of the struct with unknown amount of data. If size = 42, then it is data[42], if size = 165, then it is data[165].
I think the reason for the [] was explained in 9ad06ebb9ad6ec4bb707f2a4a352ad6c382b5cd5 for example.

I have tried all possible permutation of `DECLARE_FLEX_ARRAY` but it just does not help and when it does then the code makes no sense, gash.